public/app/features/scenes/components/layout/SceneGridLayout.test.tsx

```diff
@@ -0,0 +1,242 @@
+          children: [
+            new TestObject({ size: { x: 0, y: 0, width: 12, height: 5 } }),
+            new TestObject({ size: { x: 0, y: 5, width: 12, height: 5 } }),
+          ],
+        }),
+      });
+
+      render(<scene.Component model={scene} />);
+
+      expect(screen.queryAllByTestId('test-object')).toHaveLength(2);
+    });
+
+    it('should not render children of a collapsed row', async () => {
+      const scene = new Scene({
+        title: 'Grid test',
+        layout: new SceneGridLayout({
+          children: [
+            new TestObject({ key: 'a', size: { x: 0, y: 0, width: 12, height: 5 } }),
+            new TestObject({ key: 'b', size: { x: 0, y: 5, width: 12, height: 5 } }),
+            new SceneGridRow({
+              title: 'Row A',
+              key: 'Row A',
+              isCollapsed: true,
+              size: { y: 10 },
+              children: [new TestObject({ key: 'c', size: { x: 0, y: 11, width: 12, height: 5 } })],
+            }),
+          ],
+        }),
+      });
+
+      render(<scene.Component model={scene} />);
+
+      expect(screen.queryAllByTestId('test-object')).toHaveLength(2);
+    });
+
+    it('should  render children of an expanded row', async () => {
+      const scene = new Scene({
+        title: 'Grid test',
+        layout: new SceneGridLayout({
+          children: [
+            new TestObject({ key: 'a', size: { x: 0, y: 0, width: 12, height: 5 } }),
+            new TestObject({ key: 'b', size: { x: 0, y: 5, width: 12, height: 5 } }),
+            new SceneGridRow({
+              title: 'Row A',
+              key: 'Row A',
+              isCollapsed: false,
+              size: { y: 10 },
+              children: [new TestObject({ key: 'c', size: { x: 0, y: 11, width: 12, height: 5 } })],
+            }),
+          ],
+        }),
+      });
+
+      render(<scene.Component model={scene} />);
+
+      expect(screen.queryAllByTestId('test-object')).toHaveLength(3);
+    });
+  });
+
+  describe('when moving a panel', () => {
+    it('shoud update layout children placement and order ', () => {
+      const layout = new SceneGridLayout({
+        children: [
+          new TestObject({ key: 'a', size: { x: 0, y: 0, width: 1, height: 1 } }),
+          new TestObject({ key: 'b', size: { x: 1, y: 0, width: 1, height: 1 } }),
+          new TestObject({ key: 'c', size: { x: 0, y: 1, width: 1, height: 1 } }),
+        ],
+      });
+
+      layout.onDragStop(
+        [
+          { i: 'b', x: 0, y: 0, w: 1, h: 1 },
+          {
+            i: 'a',
+            x: 0,
+            y: 1,
+            w: 1,
+            h: 1,
+          },
+          {
+            i: 'c',
+            x: 0,
+            y: 2,
+            w: 1,
+            h: 1,
+          },
+        ],
+        // @ts-ignore
+        {},
+        { i: 'b', x: 0, y: 0, w: 1, h: 1 },
+        // @ts-ignore
+        {},
+        // @ts-ignore
+        {},
+        // @ts-ignore
```

Comment:
To avoid people thinking these TS errors need to be fixed, you can replace `@ts-ignore` with `@ts-expect-error`, which indicates we're breaking types on purpose. In this case, we want to test what happens if the panels are not completed.